Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -1022,9 +1022,12 @@ public struct RenderNodeTranslator: SemanticVisitor {
         contentCompiler: inout RenderContentCompiler
     ) -> [TaskGroupRenderSection] {
         return topics.taskGroups.compactMap { group in
-            let supportedLanguages = Set(group.directives.compactMap {
-                SupportedLanguage(from: $0, source: nil, for: bundle, in: context)?.language
-            })
+            let supportedLanguages = group.directives[SupportedLanguage.directiveName].map {
```

Comment:
What's the advantage of using `Optional#map` here vs. unwrapping the array of directives using `if let` or some other syntax? As a Swift beginner it took me a while to figure out that the receiver of the  `map` method was an `Optional<Array<BlockDirective>>` and not `Array<BlockDirective>`.

This was made more confusing because the dictionary `[]` lookup call returns an optional, not the unwrapped value, but in a silent way... the type `Optional` or the `?` operator don't appear anywhere :)

Reply:
I love all the code examples! thank you. And I agree the final version you checked in is definitely more readable than what you had before...

For me, the biggest improvement is seeing the `?` - this gives me a clue that an option value appears on that line. Earlier I was reading the call to `map` and assuming it was mapping the elements of some collection.

Like you said, it wasn't until I saw the `??` that I realized you were working with optional values. (In fact, I had to drop into the debugger to figure out what was going on!).